### ISSUE TITLE
Run code in batches/sequentially 

### ISSUE THREAD

**Issue by edehino** [ultralytics/ultralytics#1855]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

How do i run multiple training code line sequentially?

In yolov5, I can run something like this:
!python train.py .....
!python train.py .....

But when I try:
!yolo detect train data=.....

it says yolo is not recognized as a command. My jupyter notebook has a virtual environment that has ultralytics package. Also I tried running the yolo command in anaconda terminal (with the same environment activated) and it works.

Basically I want to run the yolo code in batches sequentially so I don't need to wait for one training to finish to run another.

**Comment by glenn-jocher** [comment]
Hi @todoyo. It seems like you are trying to run YOLOv8 commands sequentially in a Jupyter notebook. You can try using the `subprocess.Popen` method to run shell commands one after another. Here's an example:

```python
import subprocess

p1 = subprocess.Popen(['python', 'train.py', '...']) # first training command
p2 = subprocess.Popen(['python', 'train.py', '...']) # second training command

p1.wait() # wait for the first training command to complete
p2.wait() # wait for the second training command to complete
```
This will run both commands in parallel, but wait for the first command to complete before starting the second command.

Regarding the "yolo" command not being recognized, make sure you are using the correct command for YOLOv8 (`detect` for object detection, `train` for training, etc.) and that the correct path to the YOLOv8 script is specified.

**Comment by glenn-jocher** [comment]
@edehino another option is to use the `&` to run bash commands in the background, i.e. see https://www.maketecheasier.com/run-bash-commands-background-linux/

**Comment by Balasolina** [comment]
i'm trying to run a paython script through the mac terminal with a bash file (in the .sh extension), the program runs without error when i run it in pycharm and spider, when i run it in my .sh file it gives an error that the training file of my yolo model is missing. below i'll provide the code i'm using to run the python file and the error i get. Note: when I put other python files in the same file, I get results in the execution. thanks

<img width="914" alt="Screenshot 2023-12-16 at 17 45 44" src="https://github.com/ultralytics/ultralytics/assets/139721396/61e84d04-52b8-47aa-b184-d8e5d846342e">
<img width="546" alt="Screenshot 2023-12-16 at 17 46 10" src="https://github.com/ultralytics/ultralytics/assets/139721396/3f3a05f5-4278-4a20-a2a1-be7503ea6a62">

**Comment by glenn-jocher** [comment]
@Balasolina hello! It looks like the script is unable to locate the training file when run from the terminal, but works in IDEs like PyCharm and Spyder. This often indicates a difference in the working directory. When you run a script in an IDE, the working directory is often automatically set to the project or script folder. However, when running from a terminal, the working directory is the location from which the command is invoked.

Ensure your bash script navigates to the appropriate directory where your YOLO training file resides before running the Python script. You can do this by adding a `cd /path/to/your/directory` command before your Python script execution in your `.sh` file.

If your script relies on relative paths to find resources like the training file, running it from a different directory could cause the file not to be found. Verify that all necessary files are accessible from the directory where you're executing the script.

Lastly, always check that your environment variables and paths are properly set in the terminal the same way they are in your development environment. This includes checking that the correct Python environment is activated if using virtual environments.

**Comment by pinnintipraneethkumar** [comment]
Hi @glenn-jocher , when i run this code i can able to start the training but the training is not proceding further like it creates a directory named train in which it creates a weights directory thats it just stuck there.
import subprocess

# Your command and arguments
command = ['yolo', 'train','detect', 'patience=0']

try:
    process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)

    # Wait for the process to complete and get the return code
    return_code = process.wait()

    # Access process.stdout and process.stderr as needed
    stdout, stderr = process.communicate()

    print("Return Code:", return_code)
    print("STDOUT:", stdout)
    print("STDERR:", stderr)
except Exception as e:
    print(f"Error running subprocess: {e}")

**Comment by pinnintipraneethkumar** [comment]
Thanks @ glenn-jocher it worked !
